core/src/main/java/hudson/model/listeners/ItemListener.java

```diff
@@ -171,23 +171,8 @@ public void register() {
         return ExtensionList.lookup(ItemListener.class);
     }
 
-    // TODO JENKINS-21224 generalize this to a method perhaps in ExtensionList and use consistently from all listeners
```

Comment:
Just one example of a listener converted to the new API, in addition to `ComputerListener`. There are a lot more that could be done.